nesta/packages/crunchbase/crunchbase_collect.py

```diff
@@ -0,0 +1,333 @@
+from contextlib import contextmanager
+import logging
+import pandas as pd
+import re
+import requests
+import tarfile
+from tempfile import NamedTemporaryFile
+
+from nesta.production.luigihacks import misctools
+from nesta.production.orms.orm_utils import insert_data
+from nesta.packages.geo_utils.country_iso_code import country_iso_code_to_name
+from nesta.packages.geo_utils.geocode import generate_composite_key
+
+
+@contextmanager
+def crunchbase_tar():
+    """Downloads the tar archive of Crunchbase data.
+
+    Returns:
+        :code:`tarfile.Tarfile`: opened tar archive
+    """
+    crunchbase_config = misctools.get_config('crunchbase.config', 'crunchbase')
+    user_key = crunchbase_config['user_key']
+    url = 'https://api.crunchbase.com/v3.1/csv_export/csv_export.tar.gz?user_key='
+    with NamedTemporaryFile() as tmp_file:
+        r = requests.get(''.join([url, user_key]))
+        tmp_file.write(r.content)
+        try:
```

Comment:
From the point of view of memory management, I think I would change this to:

       with NamedTemporaryFile() as tmp_file:
            r = requests.get(''.join([url, user_key]))
            tmp_file.write(r.content)
            tmp_tar = tarfile.open(tmp_file.name)

       try:
           yield tmp_tar
       finally:
           tmp_tar.close()

to allow `tmp_file` to go out of scope (though how useful this is depends how large the `tmp_file` is and for how long `tmp_tar` will be in scope)

Reply:
temp file is huge >600MB! 

Good point, I will get this changed.

Reply:
need to add a test case to mock up a tar file

Reply:
Done!